internal/oem/oem.go

```diff
@@ -52,6 +53,10 @@ func (c Config) FetchFunc() providers.FuncFetchConfig {
 	return c.fetch
 }
 
+func (c Config) NewFetcherFunc() providers.FuncNewFetcher {
+	return c.newFetcher
```

Comment:
Instead of forcing every provider to redefine the same function, why not make `newFetcher` nullable and provide a default?